### ISSUE TITLE
L010: SQLFluff 0.9.1 capitalizes columns named date when used as parameters in DATE_ADD 

### ISSUE THREAD

**Issue by kdw2126** [sqlfluff/sqlfluff#2281]
## Expected Behaviour
When run on a SQL file containing a line like DATE_ADD(date, INTERVAL 5 YEAR), SQLFluff should not prompt or change date to be capitalized because it is a variable name, not a SQL keyword, in this context.

## Observed Behavior
Starting in SQLFluff 0.9.1, for this function (and possibly others) DATE is treated like an SQL keyword rather than a column name.

## Steps to Reproduce
(1). Generate a SQL file of the form:
```sql
SELECT DATE_ADD(date, INTERVAL 5 YEAR) AS display_date
FROM
    model
```
(2). Set BigQuery as the dialect and run sqlfluff lint on this file.

Output is **L:   1 | P:  17 | L010 | Keywords must be upper case.**

## Dialect
BigQuery

## Version
SQLFluff version is 0.9.1, testing in 0.9.0 doesn't produce the same error
Python version is 3.8.1.0.
(I'm using DBT but the same issue occurs when not using dbt as a template).

## Configuration
```
[sqlfluff]
dialect = bigquery
exclude_rules = L003,L008,L011,L014,L016,L029,L031,L034

[sqlfluff:rules]
max_line_length = 120
comma_style = leading

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
capitalisation_policy = upper
```

**Comment by jpy-git** [comment]
Good spot!

I think this is a parsing issue rather than an issue with L010.

If you parse with a more obvious column name
e.g.
```sql
SELECT DATE_ADD(abc, INTERVAL 5 YEAR) AS display_date
FROM
    model
```
the function looks like:
```
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'DATE_ADD'
[L:  1, P: 16]      |                        bracketed:
[L:  1, P: 16]      |                            start_bracket:                    '('
[L:  1, P: 17]      |                            [META] indent:
[L:  1, P: 17]      |                            expression:
[L:  1, P: 17]      |                                column_reference:
[L:  1, P: 17]      |                                    identifier:               'abc'
[L:  1, P: 20]      |                            comma:                            ','
[L:  1, P: 21]      |                            whitespace:                       ' '
[L:  1, P: 22]      |                            expression:
[L:  1, P: 22]      |                                interval_expression:
[L:  1, P: 22]      |                                    keyword:                  'INTERVAL'
[L:  1, P: 30]      |                                    whitespace:               ' '
[L:  1, P: 31]      |                                    expression:
[L:  1, P: 31]      |                                        literal:              '5'
[L:  1, P: 32]      |                                    whitespace:               ' '
[L:  1, P: 33]      |                                    date_part:                'YEAR'
[L:  1, P: 37]      |                            [META] dedent:
[L:  1, P: 37]      |                            end_bracket:                      ')'
```

whereas when you parse with the column called `date` (also `datetime`) it gives:
```
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'DATE_ADD'
[L:  1, P: 16]      |                        bracketed:
[L:  1, P: 16]      |                            start_bracket:                    '('
[L:  1, P: 17]      |                            [META] indent:
[L:  1, P: 17]      |                            date_part:                        'date'
[L:  1, P: 21]      |                            comma:                            ','
[L:  1, P: 22]      |                            whitespace:                       ' '
[L:  1, P: 23]      |                            expression:
[L:  1, P: 23]      |                                interval_expression:
[L:  1, P: 23]      |                                    keyword:                  'INTERVAL'
[L:  1, P: 31]      |                                    whitespace:               ' '
[L:  1, P: 32]      |                                    expression:
[L:  1, P: 32]      |                                        literal:              '5'
[L:  1, P: 33]      |                                    whitespace:               ' '
[L:  1, P: 34]      |                                    date_part:                'YEAR'
[L:  1, P: 38]      |                            [META] dedent:
[L:  1, P: 38]      |                            end_bracket:                      ')'
```

**Comment by jpy-git** [comment]
So `date` is being interpreted as a `date_part` rather than a `column_reference` and therefore it gets capitalised. (Introduced in #2167).

A PR should be looking at fixing the function grammar to make `date` parse as a `column_reference` in this context

**Comment by jpy-git** [comment]
Interestingly `date` already parses as a `column_reference` is set to ANSI and therefore this error doesn't occur. So this must be bigquery specific

**Comment by tunetheweb** [comment]
Noticed this myself. Raised fix in #2283.
